src/agent_handler.py

```diff
@@ -176,10 +184,25 @@ async def process_agent_run(runner, session, exit_stack, user_query, remediation
             if event_count > max_events_limit:
                 log(f"\n⚠️ Reached maximum event limit of {max_events_limit} for {agent_type.upper()} agent. Stopping agent execution early.")
                 final_response += f"\n\n⚠️ Note: Agent execution was terminated early after reaching the maximum limit of {max_events_limit} events. The solution may be incomplete."
-                await events_async.aclose()
-                # Throw exception to fully abort processing
-                error_exit(remediation_id, FailureCategory.EXCEEDED_AGENT_EVENTS.value)
+                
+                # First clean up the event stream gracefully
+                try:
+                    await _cleanup_event_stream(events_async)
+                except Exception:
+                    pass  # Ignore any cleanup errors
```

Comment:
Do we want any debug logs in these exceptions ? Or we've just determined that it's just noise at this point.